cmd/groundcover/deploy.go

```diff
@@ -68,56 +68,64 @@ var DeployCmd = &cobra.Command{
 			return err
 		}
 
-		var nodeList *v1.NodeList
-		if nodeList, err = kubeClient.CoreV1().Nodes().List(cmd.Context(), metav1.ListOptions{}); err != nil {
-			return err
-		}
-		numberOfNodes := len(nodeList.Items)
-
 		var chart *helm.Chart
 		if chart, err = helmClient.GetLatestChart(CHART_NAME, HELM_REPO_URL); err != nil {
 			return err
 		}
 
-		var isReleaseInstalled bool
-		if isReleaseInstalled, err = helmClient.IsReleaseInstalled(releaseName); err != nil {
+		var nodesSummeries []k8s.NodeSummary
+		if nodesSummeries, err = kubeClient.GetNodesSummeries(cmd.Context()); err != nil {
 			return err
 		}
+		nodesCount := len(nodesSummeries)
 
-		isUpgrade := false
-		isLatestNewer := false
-		if isReleaseInstalled {
-			var release *helm.Release
-			if release, err = helmClient.GetCurrentRelease(releaseName); err != nil {
-				return err
-			}
-			isUpgrade = true
-			isLatestNewer = chart.Version().GT(release.Version())
+		var isUpgrade bool
+		if isUpgrade, err = helmClient.IsReleaseInstalled(releaseName); err != nil {
+			return err
 		}
 
 		var promptMessage string
+		var expectedAlligatorsCount int
 		switch {
 		case !isUpgrade:
+			nodeRequirements := k8s.NewNodeMinimumRequirements()
+			adequateNodesReports, _ := nodeRequirements.GetAdequateAndInadequateNodeReports(nodesSummeries)
+			expectedAlligatorsCount = len(adequateNodesReports)
```

Comment:
if we are passing len(adequateNodesReports) as the expectedNodes to the waitForAlligator, that means we will show the user we are going to be installed only on a subset of nodes without explaining why and in potential we'll still might be installed on more nodes, I think:
- adding a check if len(adequateNodesReports) < len(nodes)  -> prompt the user with a message ( due to chart requirements not all nodes might be monitored)  or something like it
- keep this info to ourselves, waiting only for the alligators we expect to boot but hiding to node count so we wont confuse the user, wdyt?

Reply:
second option is also consistent with upgrade scenario, or else we will need to check adequate nodes there as well which is more problematic

Reply:
I suggest to use this prompt:
`fmt.Sprintf("Deploying groundcover (cluster: %s, namespace: %s, nodes: %d/%d, version: %s).\nDo you want to deploy?", clusterName, namespace, expectedAlligatorsCount, nodesCount, chart.Version(),
)`
Next PR we will need to handle inadequateNodesReports and then decide how to output the info to user